doc/user/content/ingest-data/rudderstack.md

```diff
@@ -0,0 +1,147 @@
+---
+title: "RudderStack"
+description: "How to stream data from RudderStack to Materialize using webhooks"
+menu:
+  main:
+    parent: "webhooks"
+    name: "RudderStack"
+    weight: 10
+---
+
+[RudderStack](https://rudderstack.com/) is a Customer Data Platform (CDP) for gathering and routing data from your applications.
+
+This guide will walk you through the steps to ingest data from RudderStack to Materialize.
+
+## Before you begin
+
+Ensure that you have:
+
+- [A RudderStack account](https://app.rudderstack.com/signup)
+- A RudderStack [source](https://www.rudderstack.com/docs/sources/overview/) set up and running.
+
+If you don't have a RudderStack source set up, follow the steps outlined in their [Getting Started](https://www.rudderstack.com/docs/dashboard-guides/sources/) guide.
+
+## Step 1. Create a Materialize Cluster
+
+To create a Materialize cluster, follow the steps outlined in our
+create a [cluster](/sql/create-cluster) guide,
+or create a managed cluster with two medium replicas using the following SQL statement:
+
+```sql
+CREATE CLUSTER rudderstack_cluster SIZE = '3xsmall', REPLICATION FACTOR = 2;
+```
+
+## Step 2. Create a Shared Secret
+
+To ensure data authenticity between RudderStack and Materialize, establish a shared secret:
+
+```sql
+CREATE SECRET rudderstack_shared_secret AS '<secret_value>';
+```
+
+This shared secret allows RudderStack to authenticate each incoming request, ensuring the data's integrity.
+
+Change the `<secret_value>` to a unique value that only you know and store it in a secure location.
+
+## Step 3. Set Up a Webhook Source
+
+Using the secret from the previous step, create a [webhook source](/sql/create-source/webhook/) in Materialize to ingest data from RudderStack:
+
+```sql
+CREATE SOURCE rudderstack_source IN CLUSTER rudderstack_cluster
+  FROM WEBHOOK
+    BODY FORMAT JSON
+    CHECK (
+      WITH (
+        HEADERS,
+        BODY AS request_body,
+        SECRET rudderstack_shared_secret
+      )
+      headers->'authorization' = rudderstack_shared_secret
+);
+```
+
+The above webhook source uses [basic authentication](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#basic_authentication_scheme).
+This enables a simple and rudimentary way to grant authorization to your webhook source.
+
+## Step 4. Webhook Destination in RudderStack
+
+To configure Webhook as a destination in RudderStack, follow the steps outlined below:
+
+1.  **Select Your RudderStack Source**: Identify the source you wish to add a webhook endpoint to. If you don't have a source set up, follow the steps outlined in their [Getting Started](https://www.rudderstack.com/docs/dashboard-guides/sources/) guide.
+
+1.  **Choose Webhook as Destination**:
+    - Navigate to the destination section.
+    - Select the **Webhook**" option.
+    - Assign a name to your destination.
+    - Click **Continue**.
+
+### Configuring Connection Settings
```

Comment:
Should this be another step inside the step 4?

Reply:
I also think that the inner steps should be numbered:
3. Assign a name to your destination.
4. Click **Continue.